esp/src/src-react/components/forms/Fields.tsx

```diff
@@ -101,43 +103,93 @@ const AsyncDropdown: React.FunctionComponent<AsyncDropdownProps> = ({
 
     const selOptions = React.useMemo<IDropdownOption[]>(() => {
         if (options !== undefined) {
-            return !required ? [{ key: "", text: "" }, ...options] : options;
+            return !required && !multiSelect ? [{ key: "", text: "" }, ...options] : options;
         }
         return [];
-    }, [options, required]);
+    }, [multiSelect, options, required]);
     const [selectedItem, setSelectedItem] = React.useState<IDropdownOption>();
     const [selectedIdx, setSelectedIdx] = React.useState<number>();
 
+    const [selectedItems, setSelectedItems] = React.useState<IDropdownOption[]>([]);
+
     React.useEffect(() => {
-        let item;
-        if (selectedItem?.key) {
-            item = selOptions?.find(row => row.key === selectedItem?.key) ?? selOptions[0];
+        if (multiSelect) {
+            const items: IDropdownOption[] = [];
+            if (selectedItems?.length > 0) {
+                selectedItems.forEach(item => items.push(item));
+            } else {
+                if (selectedKey === "") return;
+                const keys = selectedKey.split("|");
+                if (keys.length > 0) {
+                    keys.forEach(key => items.push({ key: key, text: key }));
+                }
+            }
+            if (!items.length) return;
+            if (JSON.stringify(items.map(item => item.key)) === JSON.stringify(selectedKey.split("|"))) {
+                // do nothing, unless
+                if (!selectedItems.length) {
+                    setSelectedItems(items);
+                }
+            } else {
+                setSelectedItems(items);
+            }
```

Comment:
Lines 127 -> 135 is equivalent to:
```ts
if (items.length && (!selectedItems.length || JSON.stringify(items.map(item => item.key)) !== JSON.stringify(selectedKey.split("|")))) {
   setSelectedItems(items);
}
```
Is that correct (I am not understanding the `!selectedItems.length` condition)?

Reply:
@jeclrsg bump?

Reply:
I did have a `!==` in there.  But it was more for me to understand why the `!selectedItem.length` was specialised - probably to do with initial load?

Reply:
That's right, it's for the initial loading of selectedItems from the url